capsules/src/mcp23008.rs

```diff
@@ -0,0 +1,546 @@
+        self.buffer.take().map_or(ReturnCode::ENOMEM, |buffer| {
+            self.i2c.enable();
+
+            buffer[0] = Registers::Gpio as u8;
+            self.i2c.write(buffer, 1);
+            self.state.set(State::SelectGpio(pin_number, value));
+
+            ReturnCode::SUCCESS
+        })
+    }
+
+    fn toggle_pin(&self, pin_number: u8) -> ReturnCode {
+        self.buffer.take().map_or(ReturnCode::ENOMEM, |buffer| {
+            self.i2c.enable();
+
+            buffer[0] = Registers::Gpio as u8;
+            self.i2c.write(buffer, 1);
+            self.state.set(State::SelectGpioToggle(pin_number));
+
+            ReturnCode::SUCCESS
+        })
+    }
+
+    fn read_pin(&self, pin_number: u8) -> ReturnCode {
+        self.buffer.take().map_or(ReturnCode::ENOMEM, |buffer| {
+            self.i2c.enable();
+
+            buffer[0] = Registers::Gpio as u8;
+            self.i2c.write(buffer, 1);
+            self.state.set(State::SelectGpioRead(pin_number));
+
+            ReturnCode::SUCCESS
+        })
+    }
+
+    fn enable_interrupt_pin(&self,
+                            pin_number: u8,
+                            direction: hil::gpio::InterruptMode)
+                            -> ReturnCode {
+        self.buffer.take().map_or(ReturnCode::ENOMEM, |buffer| {
+            self.i2c.enable();
+
+            // Mark the settings that we have for this interrupt.
+            // Since the MCP23008 only seems to support level interrupts
+            // and both edge interrupts, we choose to use both edge interrupts
+            // and then filter here in the driver if the user only asked
+            // for one direction interrupts. To do this, we need to know what
+            // the user asked for.
+            self.save_pin_interrupt_state(pin_number, true, direction);
+
+            // Setup interrupt configs that are true of all interrupts
+            buffer[0] = Registers::IntCon as u8;
+            buffer[1] = 0; // Make all pins toggle on every change.
+            buffer[2] = 0b00000010; // Make MCP23008 interrupt pin active high.
+            self.i2c.write(buffer, 3);
+            self.state.set(State::EnableInterruptSettings);
+
+            ReturnCode::SUCCESS
+        })
+    }
+
+    fn disable_interrupt_pin(&self, pin_number: u8) -> ReturnCode {
+        self.buffer.take().map_or(ReturnCode::ENOMEM, |buffer| {
+            self.i2c.enable();
+
+            // Clear this interrupt from our setup.
+            self.remove_pin_interrupt_state(pin_number);
+
+            // Just have to write the new interrupt settings.
+            buffer[0] = Registers::GpIntEn as u8;
+            buffer[1] = self.get_pin_interrupt_enabled_state();
+            self.i2c.write(buffer, 2);
+            self.state.set(State::Done);
+
+            ReturnCode::SUCCESS
+        })
+    }
+
+    /// Helper functions for keeping track of which interrupts are currently
+    /// enabled.
```

Comment:
I _think_ `///` won't do what you want here, as it'll tie this doc comment to only this function :/